### PR TITLE
Made Host && Participant Listing page similar to Main Evalai

### PR DESCRIPTION
**Pull request by Suryansh5545**
I have made both look exactly similar just with two exception i haven't removed the user icon on host page and placed radio button in middle as they look better from my point of view.
I have also added if condition for team details update function.
Selection_048
Selection_047

### CODE REVIEW

**Comment by Suryansh5545** [comment]
Sir i am unsure weather i have to made changes to the modal shown to update and add team name and email and other. or let is stay as it is

File: src/app/components/publiclists/teamlist/teamcard/teamcard.component.html [hunk 1]
```diff
@@ -1,41 +1,41 @@
...
+        {{teamView['created_by']}}
+        </span>
+      </div>
+      <div class='team-icons'>
+        <a (click)="addMembersToTeam($event)"><i class="fa fa-user-plus list-icon text-highlight w-300 left" aria-hidden="true"></i></a>
+        <a (click)="editTeam($event)"><i class="fa fa fa-pencil list-icon text-highlight w-300 left grey-text" aria-hidden="true"></i></a>
+        <a (click)="deleteTeam($event)"><i class="fa fa-trash list-icon text-highlight w-300 left grey-text" aria-hidden="true"></i></a>
+      </div>
     </div>
+      <div class='team-members'>
```
Comment (Ram81) [inline src/app/components/publiclists/teamlist/teamcard/teamcard.component.html]: @Suryansh5545 can you indent this code properly

File: src/app/components/publiclists/teamlist/teamcard/teamcard.component.scss [hunk 2]
```diff
@@ -124,3 +116,65 @@
   height: unset !important;
   border-bottom-style: none !important;
 }
+.title.toptitle {
```
Comment (Ram81) [inline src/app/components/publiclists/teamlist/teamcard/teamcard.component.scss:118]: new line above

File: src/app/components/publiclists/teamlist/teamcard/teamcard.component.scss [hunk 3]
```diff
@@ -124,3 +116,65 @@
...
+  margin-bottom: 0px !important;
+}
+.orange-text {
```
Comment (Ram81) [inline src/app/components/publiclists/teamlist/teamcard/teamcard.component.scss]: same here

File: src/app/components/publiclists/teamlist/teamcard/teamcard.component.scss [hunk 4]
```diff
@@ -124,3 +116,65 @@
...
+  color: #ff9800 !important;
+}
+.radio-button {
+  cursor: pointer;
+  padding-right: 25px;
+}
+input[type=radio] {
```
Comment (Ram81) [inline src/app/components/publiclists/teamlist/teamcard/teamcard.component.scss]: same here

File: src/app/components/publiclists/teamlist/teamlist.component.scss [hunk 5]
```diff
@@ -161,3 +161,6 @@ span.link-team-user {
     margin-top: 30px;
   }
 }
+#teamdetailslayout {
```
Comment (Ram81) [inline src/app/components/publiclists/teamlist/teamlist.component.scss]: same here

**Review by Ram81 — CHANGES_REQUESTED** [review]

**Comment by Suryansh5545** [comment]
@Ram81 Sir i have done all the changes

**Comment by Suryansh5545** [comment]
@Ram81  and Sir can you also review my this pr https://github.com/Cloud-CV/EvalAI/pull/2622

File: src/app/components/publiclists/teamlist/teamcard/teamcard.component.html [hunk 6]
```diff
@@ -1,41 +1,41 @@
...
-      </span>
-      <br>
-      <span class="title">
-	    Created by:
-      </span>
-      <span>
-	    {{teamView['created_by']}}
-      </span>
+    <div class="team-content" (click)='selectToggle()'>
+      <input type="radio"  class="radio-button__input" *ngIf='!isSelected && (isHost || isOnChallengePage)' id="644">
```
Comment (lunayach) [inline src/app/components/publiclists/teamlist/teamcard/teamcard.component.html]: Please use unique and meaningful ID's.

File: src/app/components/publiclists/teamlist/teamcard/teamcard.component.html [hunk 7]
```diff
@@ -1,41 +1,41 @@
...
+      <input type="radio"  class="radio-button__input" aria-hidden="true" *ngIf='isSelected && (isHost || isOnChallengePage)' id="644">
+      <label for="644"></label>
```
Comment (lunayach) [inline src/app/components/publiclists/teamlist/teamcard/teamcard.component.html]: @Suryansh5545 did you check the functionality when there is more than one team to choose from. We are only able to select the first team on the list. Can you please check?

**Review by lunayach — CHANGES_REQUESTED** [review]

File: src/app/components/publiclists/teamlist/teamcard/teamcard.component.html (hunk 6 above)
Comment (Suryansh5545) [reply src/app/components/publiclists/teamlist/teamcard/teamcard.component.html]: @lunayach I thought it might be good to use the same Id as used in main repo. But I will change it

Comment (Suryansh5545) [reply src/app/components/publiclists/teamlist/teamcard/teamcard.component.html]: oh Sorry Sir my mistake i didn't see the ids are generated randomly

File: src/app/components/publiclists/teamlist/teamcard/teamcard.component.html (hunk 7 above)
Comment (Suryansh5545) [reply src/app/components/publiclists/teamlist/teamcard/teamcard.component.html]: Sir It was happening because of me putting a static id . while i am not able to use value.id which was used in amin evalai repo . i have used team_name instead but it serve the purpose  well.

**Comment by Suryansh5545** [comment]
@Sanji515 @lunayach Sir is it good now?

**Comment by Sanji515** [comment]
@Suryansh5545 Please compare it with the current version like the color of fa-users icon and text color doesn't look the same. Also after selecting the team the way it show the selection on the radio is button isn't same (it shouldn't fill the entire circle).

Also it doesn't close the `collapsible body` after clicking again on the card.

**Comment by Suryansh5545** [comment]
@Sanji515 Sir, i have fixed the circle colour . but when i tried to fix the collapsible it makes the radio button irresponsible because of ng-if so how can i solve this issue

**Comment by Suryansh5545** [comment]
@Sanji515 I tried many thing but i am still unable to solve this bug please help me.

**Comment by Sanji515** [comment]
> @Sanji515 I tried many thing but i am still unable to solve this bug please help me.

Hey @Suryansh5545 It's not toggling for the host teams, please try to look at the `selectToggle` function over [here](https://github.com/Cloud-CV/EvalAI-ngx/blob/master/src/app/components/publiclists/teamlist/teamcard/teamcard.component.ts#L101) if its working correctly :smile:

**Comment by Suryansh5545** [comment]
@Sanji515 I did tried that sir . But it brokes the radio button and then I can't click it.

**Comment by Sanji515** [comment]
> @Sanji515 I did tried that sir . But it brokes the radio button and then I can't click it.

yes, you need to modify that function in such a way that it should toggle on every call to that function but select the team only when the user is host.

**Comment by Suryansh5545** [comment]
@Sanji515 Sir,
I have implemented  the fix for the collapsible and the function to select team also works fine but the radio button still doesn't get selected. I have pushed my code

File: src/app/components/publiclists/teamlist/teamcard/teamcard.component.html [hunk 8]
```diff
@@ -1,41 +1,41 @@
...
-	    {{teamView['team_name']}}
-      </span>
-      <br>
-      <span class="title">
-	    Created by:
-      </span>
-      <span>
-	    {{teamView['created_by']}}
-      </span>
+    <div class="team-content" (click)='selectToggle()'>
```
Comment (Sanji515) [inline src/app/components/publiclists/teamlist/teamcard/teamcard.component.html:3]: Hey @Suryansh5545 I think this is causing the problem. You need to add this `on click` excluding the radio inputs.

Comment (Suryansh5545) [reply src/app/components/publiclists/teamlist/teamcard/teamcard.component.html:3]: @Sanji515  i cant understand sir do i have to add this to my radio button?

Comment (Sanji515) [reply src/app/components/publiclists/teamlist/teamcard/teamcard.component.html:3]: Nope, not on to your radio buttons but excluding them.

Comment (Suryansh5545) [reply src/app/components/publiclists/teamlist/teamcard/teamcard.component.html:3]: So the function shouldn't apply to radio button but to everything else?

Comment (Sanji515) [reply src/app/components/publiclists/teamlist/teamcard/teamcard.component.html]: Nope, not on to your radio buttons but excluding them.

Comment (Suryansh5545) [reply src/app/components/publiclists/teamlist/teamcard/teamcard.component.html:3]: @Sanji515 Sir like i have created two function one is executed when the team-content is pressed which will make it open and close and other function executes when the radio button is pressed so the team can be selected . So the function selectToggle isn't applying to the radio buttons already.

Comment (Sanji515) [reply src/app/components/publiclists/teamlist/teamcard/teamcard.component.html:3]: > Nope, not on to your radio buttons but excluding them.

I'm saying this cause in `selectToggle` function you are just toggling the selection class but not calling `selectTeam` function.

But now I think we don't need to do that, you just need to remove the extra radio type input and its condition [here](https://github.com/Cloud-CV/EvalAI-ngx/pull/276/files#diff-1343c36995ebad881639aeba7d3a10e3R4)

Comment (Suryansh5545) [reply src/app/components/publiclists/teamlist/teamcard/teamcard.component.html:3]: @Sanji515  I have done the changes sir.

**Comment by Suryansh5545** [comment]
@Sanji515 Sir is this pr ready or more work is needed?

File: src/app/components/publiclists/teamlist/teamcard/teamcard.component.html [hunk 9]
```diff
@@ -1,9 +1,6 @@
 <div class="team-card-container" [class.selected]='isSelected' [class.noexpand] = 'isHost' [class.expand] = '!isHost'>
 <div class="team-card-container-child">
     <div class="team-content" (click)='selectToggle()'>
-      <input type="radio" class="with-gap selectPhase" (click)='selectTeamToggle()' *ngIf='!isSelected && (isHost || isOnChallengePage)' id="{{teamView['team_name']}}">
-      <input type="radio" class="with-gap selectPhase" (click)='selectTeamToggle()' aria-hidden="true" *ngIf='isSelected && (isHost || isOnChallengePage)' id="{{teamView['team_name']}}">
-      <label for="{{teamView['team_name']}}"></label>
```
Comment (Sanji515) [inline src/app/components/publiclists/teamlist/teamcard/teamcard.component.html]: @Suryansh5545 Why have you removed both inputs?

File: package.json [hunk 10]
```diff
@@ -32,7 +32,7 @@
     "@types/moment-timezone": "^0.5.12",
     "angular-froala-wysiwyg": "^3.0.0-rc.2-1",
     "core-js": "^2.4.1",
-    "materialize-css": "^0.97.8",
+    "materialize-css": "^1.0.0",
```
Comment (Sanji515) [inline package.json]: Why this change here?

File: package.json [hunk 11]
```diff
@@ -41,7 +41,7 @@
     "normalize.css": "^8.0.0",
     "rxjs": "^6.5.2",
     "tslib": "^1.9.0",
-    "zone.js": "^0.8.29"
+    "zone.js": "^0.10.2"
```
Comment (Sanji515) [inline package.json]: Same why this change?

File: src/app/components/publiclists/teamlist/teamlist.component.html [hunk 12]
```diff
@@ -1,6 +1,6 @@
 <section class="ev-sm-container ev-view ">
   <div class="row">
-    <div class="col s12 m6">
+    <div  class="col s12 m6">
```
Comment (Sanji515) [inline src/app/components/publiclists/teamlist/teamlist.component.html]: Please revert this change.

File: package.json (hunk 10 above)
Comment (Suryansh5545) [reply package.json]: i kind of don't know why they are here i am removing them sorry

File: src/app/components/publiclists/teamlist/teamcard/teamcard.component.html (hunk 9 above)
Comment (Suryansh5545) [reply src/app/components/publiclists/teamlist/teamcard/teamcard.component.html]: i think i misunderstood what you said before sir .

Comment (Suryansh5545) [reply src/app/components/publiclists/teamlist/teamcard/teamcard.component.html]: if i am right i only have to remove the condition?

Comment (Sanji515) [reply src/app/components/publiclists/teamlist/teamcard/teamcard.component.html]: right, but if you remove the condition do you think you need another input there? so you can remove that extra input also.

Comment (Suryansh5545) [reply src/app/components/publiclists/teamlist/teamcard/teamcard.component.html]: i have done the changes Sir please have a look.

File: src/app/components/publiclists/teamlist/teamcard/teamcard.component.html [hunk 13]
```diff
@@ -1,41 +1,40 @@
...
-      </span>
-      <br>
-      <span class="title">
-	    Created by:
-      </span>
-      <span>
-	    {{teamView['created_by']}}
-      </span>
+    <div class="team-content" (click)='selectToggle()'>
+      <input type="radio" class="with-gap selectPhase" (click)='selectTeamToggle()' id="{{teamView['team_name']}}">
```
Comment (Sanji515) [inline src/app/components/publiclists/teamlist/teamcard/teamcard.component.html]: I think you also need to add `name` attribute otherwise it will not toggle the selection.

File: src/app/components/publiclists/teamlist/teamcard/teamcard.component.html (hunk 9 above)
Comment (Suryansh5545) [reply src/app/components/publiclists/teamlist/teamcard/teamcard.component.html]: @Sanji515 Sir the extra input is there because the id of the radio button shown is dynamic which will make both of them checkable.

**Comment by Sanji515** [comment]
@Suryansh5545 Please take at the current version like the color of fa-users icon and text color doesn't look same.

File: src/app/components/publiclists/teamlist/teamcard/teamcard.component.html (hunk 13 above)
Comment (Suryansh5545) [reply src/app/components/publiclists/teamlist/teamcard/teamcard.component.html]: Thx Sir it solved the problem

**Comment by Suryansh5545** [comment]
@Sanji515 Sir i have made the text and user icon look same. but i am having trouble changing the colour of text from orange to normal in host team page

**Comment by Sanji515** [comment]
> @Sanji515 Sir i have made the text and user icon look same. but i am having trouble changing the colour of text from orange to normal in host team page

Can you please tell which text you are talking about?

**Comment by Suryansh5545** [comment]
@Sanji515 This one sir on host page
Selection_050
Selection_049

**Comment by Sanji515** [comment]
What problem you are facing in it? It will be a link with orange color if `team url` is present else it will be a normal text.

**Comment by Suryansh5545** [comment]
oh so i dont have to implement or change anything in text colour?

**Comment by Suryansh5545** [comment]
i though it was hardcoded in participate team tab so i did same

**Comment by Suryansh5545** [comment]
@Sanji515 i have changed the colour back to normal

**Comment by Suryansh5545** [comment]
@Sanji515 Sir Should i submit the task if pr is ready?

File: src/app/components/publiclists/teamlist/teamcard/teamcard.component.html [hunk 14]
```diff
@@ -1,41 +1,40 @@
...
-      <span class="title">
-	    Created by:
-      </span>
-      <span>
-	    {{teamView['created_by']}}
-      </span>
+    <div class="team-content" (click)='selectToggle()'>
+      <input type="radio" class="with-gap selectPhase" name="TeamSelection" (click)='selectTeamToggle()' id="{{teamView['team_name']}}">
+      <label for="{{teamView['team_name']}}"></label>
+      <i class="fa fa-users" aria-hidden="true"></i>
```
Comment (Sanji515) [inline src/app/components/publiclists/teamlist/teamcard/teamcard.component.html]: In the current version we don't have user icon for host team, can you please look into this?

Comment (Suryansh5545) [reply src/app/components/publiclists/teamlist/teamcard/teamcard.component.html]: yea i saw that but i though it would look much better. but i will remove it now

File: src/app/components/publiclists/teamlist/teamcard/teamcard.component.html [hunk 15]
```diff
@@ -1,41 +1,40 @@
...
-      </span>
-      <br>
-      <span class="title">
-	    Created by:
-      </span>
-      <span>
-	    {{teamView['created_by']}}
-      </span>
+    <div class="team-content" (click)='selectToggle()'>
+      <input type="radio" class="with-gap selectPhase" name="TeamSelection" (click)='selectTeamToggle()' id="{{teamView['team_name']}}">
```
Comment (Sanji515) [inline src/app/components/publiclists/teamlist/teamcard/teamcard.component.html]: Ah forgot, need to add the condition if it is host then only show the input.

File: src/app/components/publiclists/teamlist/teamcard/teamcard.component.html [hunk 16]
```diff
@@ -1,41 +1,40 @@
...
+        Created by:
+        </span>
+        <span>
+        {{teamView['created_by']}}
+        </span>
+      </div>
+      <div class='team-icons'>
+        <a (click)="addMembersToTeam($event)"><i class="fa fa-user-plus list-icon text-highlight w-300 left" aria-hidden="true"></i></a>
+        <a (click)="editTeam($event)"><i class="fa fa fa-pencil list-icon text-highlight w-300 left grey-text" aria-hidden="true"></i></a>
+        <a (click)="deleteTeam($event)"><i class="fa fa-trash list-icon text-highlight w-300 left grey-text" aria-hidden="true"></i></a>
```
Comment (Sanji515) [inline src/app/components/publiclists/teamlist/teamcard/teamcard.component.html:28]: Minor: Can you please do the same spacing as in the current version?

File: src/app/components/publiclists/teamlist/teamlist.component.scss [hunk 17]
```diff
@@ -161,3 +161,7 @@ span.link-team-user {
     margin-top: 30px;
   }
 }
+
+#teamdetailslayout {
+  margin-left: 40px;
```
Comment (Sanji515) [inline src/app/components/publiclists/teamlist/teamlist.component.scss]: Why this change?

Comment (Suryansh5545) [reply src/app/components/publiclists/teamlist/teamlist.component.scss]: Just a part of old experiment . i removed it

File: src/app/components/publiclists/teamlist/teamlist.component.ts [hunk 18]
```diff
@@ -393,6 +393,23 @@ export class TeamlistComponent implements OnInit, OnDestroy {
   editTeamWrapper() {
     const SELF = this;
     const editTeam = (team) => {
+      if (this.isHost) {
+        SELF.apiCall = (params) => {
+          const BODY = JSON.stringify(params);
+          SELF.apiService.patchUrl(SELF.endpointsService.HostTeamURL(team), BODY).subscribe(
```
Comment (Sanji515) [inline src/app/components/publiclists/teamlist/teamlist.component.ts]: Here we can just update the url based on the condition instead of whole API call.

File: src/app/components/publiclists/teamlist/teamcard/teamcard.component.html (hunk 16 above)
Comment (Suryansh5545) [reply src/app/components/publiclists/teamlist/teamcard/teamcard.component.html:28]: Sir  space is same as current version

File: src/app/components/publiclists/teamlist/teamlist.component.ts (hunk 18 above)
Comment (Suryansh5545) [reply src/app/components/publiclists/teamlist/teamlist.component.ts]: Done

**Comment by Suryansh5545** [comment]
@Sanji515 Sir all changes made

**Comment by Sanji515** [comment]
- On clicking the card it should close the previous collapsible body if found open (as in the current version).

- On the current version of participant teams page there is some padding before the (black color) users icon and also please remove extra space between the radio input (circle) and the text `Team:`. Try to compare it with the current version.

- Please fix the alignment of team members section which is under collapsible body.

@Suryansh5545 Please take a look at these(^)

**Comment by Suryansh5545** [comment]
@Sanji515 Sir i have fixed the margins. but i am unable to make the collapsible close when another is opened. please help

**Comment by Suryansh5545** [comment]
i need some kind of hint on how to do this i tried using if statement to made isSelected false but it affects other card not the one opened before

**Comment by Suryansh5545** [comment]
@Sanji515 Sir i have added node-sass in package.json as the build fails without it . but i dont know why it failed i thought i need node-sass only for ng serve to work .

**Comment by Sanji515** [comment]
> @Sanji515 Sir i have fixed the margins. but i am unable to make the collapsible close when another is opened. please help

Can you please take reference from the master branch as it was working perfectly for the host teams

**Review by ShivamRai2003 — APPROVED** [review]

**Comment by Suryansh5545** [comment]
@Sanji515 Sir i have fix the issue . please take a look at weather my approach is good or not.
Also Sir only 1 hour is left after which i cant submit task so . if there is any mistake please let me know.

**Comment by Suryansh5545** [comment]
@Sanji515 Sir for some reason the travis status is not updating? but it has passed

**Comment by Suryansh5545** [comment]
@Sanji515 Sir I have forced pushed the repo to make the travis status change

File: src/app/components/publiclists/teamlist/teamcard/teamcard.component.scss [hunk 19]
```diff
@@ -124,3 +114,72 @@
...
+.radio-button__input:checked+.radio-button__control {
+    border-color: #01a982;
+}
+
+.radio-button__control {
+  transform: scale(0.75);
+}
+.fa-users {
+  color: black;
+  padding-right: 10px;
```
Comment (Sanji515) [inline src/app/components/publiclists/teamlist/teamcard/teamcard.component.scss]: Can we please remove this `padding-right` and add `padding: 0px 8px` for consistency purpose?

Comment (Suryansh5545) [reply src/app/components/publiclists/teamlist/teamcard/teamcard.component.scss]: sure

Comment (Suryansh5545) [reply src/app/components/publiclists/teamlist/teamcard/teamcard.component.scss]: done sir

File: package.json [hunk 20]
```diff
@@ -61,6 +61,7 @@
     "karma-coverage-istanbul-reporter": "^1.2.1",
     "karma-jasmine": "~1.1.0",
     "karma-jasmine-html-reporter": "^0.2.2",
+    "node-sass": "^4.13.1",
```
Comment (Sanji515) [inline package.json]: Please revert this change as this is not related to this PR.

Comment (Suryansh5545) [reply package.json]: Sir as i mentioned above the build is failing without this so i need to add it

Comment (Suryansh5545) [reply package.json]: https://travis-ci.org/Cloud-CV/EvalAI-ngx/builds/640847824?utm_source=github_status&utm_medium=notification

File: src/app/components/publiclists/teamlist/teamcard/teamcard.component.html [hunk 21]
```diff
@@ -1,41 +1,40 @@
...
-      </span>
-      <br>
-      <span class="title">
-	    Created by:
-      </span>
-      <span>
-	    {{teamView['created_by']}}
-      </span>
+    <div class="team-content" (click)='selectToggle()'>
+      <input type="radio" class="with-gap selectPhase" name="TeamSelection" (click)='selectTeamToggle()' *ngIf="isHost" id="{{teamView['team_name']}}">
```
Comment (Sanji515) [inline src/app/components/publiclists/teamlist/teamcard/teamcard.component.html]: Please camel case the name attribute.

File: src/app/components/publiclists/teamlist/teamcard/teamcard.component.html [hunk 22]
```diff
@@ -1,41 +1,40 @@
...
+      <label id="TeamToggleButton" for="{{teamView['team_name']}}"></label>
```
Comment (Sanji515) [inline src/app/components/publiclists/teamlist/teamcard/teamcard.component.html]: Same here for `id`.

File: src/app/components/publiclists/teamlist/teamcard/teamcard.component.html (hunk 21 above)
Comment (Suryansh5545) [reply src/app/components/publiclists/teamlist/teamcard/teamcard.component.html]: Done sir

File: src/app/components/publiclists/teamlist/teamcard/teamcard.component.html (hunk 22 above)
Comment (Suryansh5545) [reply src/app/components/publiclists/teamlist/teamcard/teamcard.component.html]: Done sir

File: package.json (hunk 20 above)
Comment (Suryansh5545) [reply package.json]: I have removed it lets see if build is successful

Comment (Suryansh5545) [reply package.json]: @Sanji515 Sir the build is failing without it. so should i add it?

**Comment by Sanji515** [comment]
> @Sanji515 Sir i have fix the issue . please take a look at weather my approach is good or not.
> Also Sir only 1 hour is left after which i cant submit task so . if there is any mistake please let me know.

Hey @Suryansh5545 I think the previous implementation was more better but you were making a mistake by not the wrapper on the `selectToggle` method instead you were calling on `selectTeamToggle` and inside this method it calls the function `selectTeam`.

**Comment by Sanji515** [comment]
Also @Suryansh5545  can you please change spacing between the icons [here](https://github.com/Cloud-CV/EvalAI-ngx/pull/276/files#diff-4ba9dd132741315811e587b427ca726cR56) to `margin: 0px 14px 0px 14px;` and also please change the container padding [here](https://github.com/Cloud-CV/EvalAI-ngx/blob/master/src/styles/base.scss#L1784) to `padding: 10px 35px;` to look more consistent with the current version.

**Comment by Suryansh5545** [comment]
@Sanji515 Sir if i put the selectteam function in the selectToggle then it will select is and the radio button will be technically useless. also i have removed the selectedteam from unselectotherteam function call from selectTeamWrapper as the radio button shouldn't open or close it.

File: package.json (hunk 20 above)
Comment (Suryansh5545) [reply package.json]: @Sanji515 Sir should i add it?

**Comment by Sanji515** [comment]
> @Sanji515 Sir if i put the selectteam function in the selectToggle then it will select is and the radio button will be technically useless. also i have removed the selectedteam from unselectotherteam function call from selectTeamWrapper as the radio button shouldn't open or close it.

okay then how about if we just send variable equal to **null** when we click on the card leaving the radio input and send **selected team object** when we click on the input button.

**Comment by Suryansh5545** [comment]
@Sanji515 Sir sending a null variable instead of this.team throws an error `Cannot read property 'id' of undefined`

**Comment by Sanji515** [comment]
> @Sanji515 Sir sending a null variable instead of this.team throws an error `Cannot read property 'id' of undefined`

yup, you have to deal the case of `null` on that function also.

**Comment by Suryansh5545** [comment]
Sir so we have to show next button even if the value is null right?

**Comment by Suryansh5545** [comment]
@Sanji515 Sir sending null and then checking for it would open both the cards as it doesn't have the id of which one to close or open so this approach might not be possible

**Comment by Sanji515** [comment]
> @Sanji515 Sir sending null and then checking for it would open both the cards as it doesn't have the id of which one to close or open so this approach might not be possible

hmm I see.... then with the team object you can send one extra boolean variable which will indicate whether it clicked on the radio input button or leaving the button.

**Comment by Suryansh5545** [comment]
@Sanji515 Sir as emit can carry one value only i am trying to use this approach `    this.selectTeamCard.emit({team: this.team, radioButton: this.isButtonChecked});` for them to carry them both but i cant call the in teamlist.

**Comment by Suryansh5545** [comment]
@Sanji515 Sir how can I call two emits value in the function??

**Comment by Suryansh5545** [comment]
@Sanji515 Sir should I continue with implementing my previous approach as this one isn't possible because emit can contain only 1 value.

**Comment by Suryansh5545** [comment]
@Sanji515 Sir which approach should I take?

**Comment by Suryansh5545** [comment]
@RishabhJain2018 Sir should I implement my previous approach??

**Comment by Suryansh5545** [comment]
@Sanji515 Sir, As GCI has ended and I am no longer working on this So I am closing the pr and I think this can be repurposed for Gsoc.

**Comment by Sanji515** [comment]
> @Sanji515 Sir, As GCI has ended and I am no longer working on this So I am closing the pr and I think this can be repurposed for Gsoc.

@Suryansh5545 As you have given you a lot of effort in this and it is almost done, if you can continue working on this then we can merge this PR.

**Comment by Suryansh5545** [comment]
@Sanji515 I would be very happy to work but as my final exams stand on my gates . I would be unable to work until 6 march.
